packages/xo-web/src/xo-app/xoa/update/index.js

```diff
@@ -84,23 +87,51 @@ const Updates = decorate([
     initialState: () => ({
       ...initialProxyState(),
       ...initialRegistrationState(),
+      ...initialChannels(),
       askRegisterAgain: false,
+      xoaReleaseChannels: [],
       showPackagesList: false,
     }),
     effects: {
       async configure() {
-        await xoaUpdater.configure(
-          pick(this.state, [
+        await xoaUpdater.configure({
+          ...pick(this.state, [
             'proxyHost',
             'proxyPassword',
             'proxyPort',
             'proxyUser',
-          ])
-        )
+          ]),
+          channel: this.state.addPrivateChannel
+            ? this.state.privateChannelId
+            : this.state.channelId,
+        })
         return this.effects.resetProxyConfig()
       },
-      initialize() {
-        return this.effects.update()
+      async initialize() {
+        this.effects.update()
+        await this.effects.getReleaseChannels()
+        this.effects.initializeChannel()
+      },
+      initializeChannel() {
+        const { xoaConfiguration } = this.props
+        if (xoaConfiguration.channel === undefined) {
+          this.state.channelId = ''
+        } else {
+          if (
+            // Is public channel
+            find(
+              this.state.xoaReleaseChannels,
+              channel => channel.id === xoaConfiguration.channel
+            )
+          ) {
+            return { channelId: xoaConfiguration.channel }
```

Comment:
I think it's weird mixing reducer syntax and setting the state directly.

Use one or the other in a given effect.